### PR TITLE
Add `useTtl` option to a file system cache

### PR DESCRIPTION
**Pull request by better-salmon**
Experiment TTL for the file system cache

### CODE REVIEW

**Comment by better-salmon** [comment]
@MauriceArikoglu check out a prototype of TTL feature for the file system

File: packages/cache-handler/src/cache-handler.ts [hunk 1]
```diff
@@ -456,24 +486,32 @@
...
-            const bodyFileData = await fsPromises.readFile(bodyFilePath);
             const { mtime } = await fsPromises.stat(bodyFilePath);
 
+            const lastModified = mtime.getTime();
+
+            if (isStale(lastModified, maxAgeSeconds, IncrementalCache.#useTtl)) {
+                return null;
+            }
+
+            const bodyFileData = await fsPromises.readFile(bodyFilePath);
```
[the remark this answers is not in the document]
Comment (MauriceArikoglu) [reply packages/cache-handler/src/cache-handler.ts]: Not a problem in this case

File: packages/cache-handler/src/cache-handler.ts [hunk 2]
```diff
@@ -456,24 +486,33 @@
...
             const bodyFilePath = this.#getFilePath(`${cacheKey}.body`, 'app');
 
-            const bodyFileData = await fsPromises.readFile(bodyFilePath);
             const { mtime } = await fsPromises.stat(bodyFilePath);
 
+            if (isStale(mtime.getTime(), maxAgeSeconds, IncrementalCache.#useTtl)) {
+                return null;
+            }
+
+            const bodyFileData = await fsPromises.readFile(bodyFilePath);
```
[the remark this answers is not in the document]
Comment (MauriceArikoglu) [reply packages/cache-handler/src/cache-handler.ts]: Not a problem in this case

File: apps/cache-testing/cache-handler-redis-stack.mjs [hunk 3]
```diff
@@ -16,6 +16,12 @@ IncrementalCache.onCreation(async () => {
 
     client.on('error', () => {});
 
+    function useTtl(maxAge) {
+        const evictionAge = maxAge * 1.5;
```
Comment (MauriceArikoglu) [inline apps/cache-testing/cache-handler-redis-stack.mjs:20]: Why *1.5?

File: packages/cache-handler/src/cache-handler.ts [hunk 4]
```diff
@@ -16,7 +16,19 @@ import type {
...
+
+type PrerenderManifest = {
+    version: 4;
+    routes: Record<
+        string,
+        {
+            initialRevalidateSeconds: number | false;
+        }
+    >;
+};
```
Comment (MauriceArikoglu) [inline packages/cache-handler/src/cache-handler.ts]: I assume this is where we store our parsed results?

Comment (better-salmon) [reply packages/cache-handler/src/cache-handler.ts]: This file is created by Next.js during the building process

**Pull request closed without merging**